tests/Dockerfile

```diff
@@ -1,9 +1,51 @@
-FROM buildkite/plugin-tester:latest
+FROM bash:4.2
```

Comment:
Is this forking away from the official Buildkite image our best option?

If we're specifically needing the older version of Bash it looks like the version to use may be configured via the `_BASH_*` environment variables built into the [bats/bats Docker image](https://hub.docker.com/layers/bats/bats/latest/images/sha256-97d91ee0aa9771e696cdf44c2b1672af484fd846eaf52ba2db6061f5c78a89d5?context=explore) the Buildkite image [is based on](https://github.com/buildkite-plugins/buildkite-plugin-tester/blob/master/Dockerfile#L1).

Reply:
If we want to control the Bash version, we have to. The environment variable you're referencing is from the [library/bash Dockerfile](https://github.com/tianon/docker-bash/blob/2d420e498d46e7241dd16e3185c8696a1f8eef36/4.2/Dockerfile) - it's only used during the bash image build and not at runtime by the entrypoint.